src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/ByteArrayMutatorFactory.java

```diff
@@ -130,7 +133,94 @@ public byte[] mutate(byte[] value, PseudoRandom prng) {
 
     @Override
     public byte[] crossOver(byte[] value, byte[] otherValue, PseudoRandom prng) {
-      return value;
+      // Passed in values are expected to already honor the min/max length constraints.
+      // As there does not seem to be an easy way to call libFuzzer's internal cross over
+      // algorithm, it is re-implemented in native Java. The algorithm is based on:
+      // https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/fuzzer/FuzzerMutate.cpp#L440
+      // https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/fuzzer/FuzzerCrossOver.cpp#L19
+      //
+      // TODO: Measure if this is fast enough.
+
+      byte[] out = null;
+      while (out == null) {
+        switch (prng.indexIn(3)) {
+          case 0:
+            out = intersect(value, otherValue, prng);
+            break;
+          case 1:
+            out = insertPart(value, otherValue, prng);
+            break;
+          case 2:
+            out = overwritePart(value, otherValue, prng);
+            break;
+          default:
+            throw new AssertionError("Invalid cross over function.");
+        }
+      }
+      return enforceLength(out);
+    }
+
+    private static byte[] intersect(byte[] value, byte[] otherValue, PseudoRandom prng) {
+      int maxOutSize = prng.closedRange(0, Math.min(value.length, otherValue.length));
+      byte[] out = new byte[maxOutSize];
+      int outPos = 0;
+      int valuePos = 0;
+      int otherValuePos = 0;
+      boolean usingFirstValue = true;
+      while (
+          outPos < out.length && (valuePos < value.length || otherValuePos < otherValue.length)) {
+        if (usingFirstValue && valuePos < value.length) {
+          int extraSize = rndArraycopy(value, valuePos, out, outPos, prng);
+          outPos += extraSize;
+          valuePos += extraSize;
+        } else if (!usingFirstValue && otherValuePos < otherValue.length) {
+          int extraSize = rndArraycopy(otherValue, otherValuePos, out, outPos, prng);
+          outPos += extraSize;
+          otherValuePos += extraSize;
+        }
+        usingFirstValue = !usingFirstValue;
+      }
+      return out;
+    }
+
+    private static int rndArraycopy(
+        byte[] val, int valPos, byte[] out, int outPos, PseudoRandom prng) {
+      int outSizeLeft = out.length - outPos;
+      int inSizeLeft = val.length - valPos;
+      int maxExtraSize = Math.min(outSizeLeft, inSizeLeft);
+      int extraSize = prng.indexIn(maxExtraSize);
+      System.arraycopy(val, valPos, out, outPos, extraSize);
+      return extraSize;
+    }
+
+    private static byte[] insertPart(byte[] value, byte[] otherValue, PseudoRandom prng) {
+      // TODO: Find some meaningful checks to do here.
+      if (value.length >= Integer.MAX_VALUE / 2) {
+        // value is already too big to be extended, try other cross over.
```

Comment:
Exactly, it will return `null` and another cross over "action" is picked. As `intersect` and `overwrite` will not increase the array size, the loop should stop eventually.